src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -3061,21 +3060,18 @@ bool sockinfo_tcp::is_readable(uint64_t *p_poll_sn, fd_array_t* p_fd_array)
 				ring* p_ring =  rx_ring_iter->first;
 				//g_p_lwip->do_timers();
 				ret = p_ring->poll_and_process_element_rx(p_poll_sn, p_fd_array);
-				if (m_n_rx_pkt_ready_list_count)
-					goto noblock;
+				if (m_n_rx_pkt_ready_list_count) {
+					m_rx_ring_map_lock.unlock();
+					return true;
+				}
 				if (ret <= 0)
 					break;
 			}
 		}
-	}
-	if (!m_n_rx_pkt_ready_list_count) {
 		m_rx_ring_map_lock.unlock();
-		return false;
 	}
-noblock:
-	m_rx_ring_map_lock.unlock();
-noblock_nolock:
-	return true;
+
+	return m_n_rx_pkt_ready_list_count ? true : false;
```

Comment:
could you consider wrapping with (...) to improve visibility (it is not mandatory)